src/modules/27-platform/auth-settings/pages/Configuration/EnablePublicAccess/EnablePublicAccess.tsx

```diff
@@ -28,29 +30,47 @@ const EnablePublicAccess: React.FC<EnablePublicAccessProps> = ({ enabled, refetc
   const [loading, setLoading] = useState(false)
   const { accountId } = useParams<AccountPathProps>()
 
-  const onChange = async (e: React.FormEvent<HTMLInputElement>): Promise<void> => {
-    const enable = e.currentTarget.checked
+  const { openDialog: openEnablePublicAccessDialog } = useConfirmationDialog({
+    contentText: enabled
+      ? getString('platform.authSettings.publicAccess.resourcesNoLongerAccessible')
+      : getString('platform.authSettings.publicAccess.resourcesWillBeAccessible'),
+    titleText: enabled
+      ? getString('platform.authSettings.publicAccess.modalTitleDisable')
+      : getString('platform.authSettings.publicAccess.modalTitleEnable'),
+    confirmButtonText: enabled ? getString('common.disable') : getString('enable'),
+    cancelButtonText: getString('cancel'),
+    intent: Intent.DANGER,
+    buttonIntent: Intent.DANGER,
+    onCloseDialog: async didConfirm => {
+      if (didConfirm) {
+        const newEnableValue = !enabled
 
-    try {
-      setLoading(true)
-      const res = await setPublicAccessPromise({
-        queryParams: {
-          accountIdentifier: accountId
-        },
-        body: enable
-      })
-      if (res.resource) {
-        // res.resource is expected as "true" if operation is successful—in both "enable" or "disable" cases
-        showSuccess(getString('platform.authSettings.updatedPublicAccess'))
-        refetchAuthSettings()
-      } else {
-        showError(defaultTo(res.responseMessages?.[0]?.message, getString('somethingWentWrong')))
+        try {
+          setLoading(true)
+          const res = await setPublicAccessPromise({
+            queryParams: {
+              accountIdentifier: accountId
+            },
+            body: newEnableValue
+          })
+          if (res.resource) {
+            // res.resource is expected as "true" if operation is successful—in both "enable" or "disable" cases
+            showSuccess(getString('platform.authSettings.publicAccess.updatedPublicAccess'))
+            refetchAuthSettings()
+          } else {
+            showError(defaultTo(res.responseMessages?.[0]?.message, getString('somethingWentWrong')))
```

Comment:
Not an issue, but just curious why the use of `defaultTo` if the same thing can be achieved with
```suggestion
            showError(res.responseMessages?.[0]?.message || getString('somethingWentWrong'))
```

Reply:
Istanbul considers `||` as a branch and gives coverage issues.